src/graph.py

```diff
@@ -0,0 +1,162 @@
+"""Implement a class for a graph data structure."""
+
+
+class Graph(object):
+    """Graph class."""
+
+    def __init__(self):
+        """Initialize a graph."""
+        self._graph = {}
+
+    def nodes(self):
+        """Return a list of all nodes in graph."""
+        return list(self._graph.keys())
+
+    def edges(self):
+        """Return a list of edges in graph."""
+        edges = []
+        for key in self._graph:
+            for i in self._graph[key]:
+                edges.append((key, i, self._graph[key][i]))
+        return edges
+
+    def add_node(self, val):
+        """Add a node with value of val to graph."""
+        self._graph[val] = {}
```

Comment:
So if I add a node with the same name as a pre-existing node, I'm going to end up overwriting it and emptying its contents